### DISCUSSION TITLE
Cross-referencing Python objects

### DISCUSSION THREAD

**Discussion by StefanoFioravanzo** [executablebooks/meta#919] [Q&A] [answered]
Hi! I was wondering if it's possible to write python API cross-references in MyST similarly to the commands described in this sphinx doc: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects

Using those Sphinx commands it's straightforward to reference a Python function or class from any page, thus creating a hyperlink to the API doc.

I couldn't find a way to make this happen in MyST, any help?

**Comment by NickleDave** [comment, marked as answer]
I could not find good docs on this either but I have found that if you remove the `:py:` and rewrite the rest of an in-line role as a MyST directive then it will work, e.g. `:py:class` -> `{class}`, `:py:func` -> `{func}`

**Reply by StefanoFioravanzo** [reply]
@NickleDave Wow thanks! That did the trick. 
out of scope since this is probably a Sphinx question: if I have to reference a class that's nested within a few packages, e.g. 

```
blablabla {class}`package1.package2.module.class_name` blablabla
```

do you know how to display the hyperlink text just as `class_name` or `module.class_name`?

**Reply by NickleDave** [reply]
That one is more complicated I think (but someone who knows better feel free to correct me).

The issue is you need autodoc to achieve what you want, specifically the autoaclass directive
https://stackoverflow.com/questions/22096187/how-to-make-sphinx-respect-importing-classes-into-package-with-init-py

but IIUC, MyST can't quite handle autodoc at the moment.
see discussion in this issue and a proof-of-concept implementation in the linked comment, that you might adapt to your project:
https://github.com/executablebooks/MyST-Parser/issues/228#issuecomment-1003634938

but I haven't tested myself whether `{autoclass}` would just automagically work the same way

**Reply by StefanoFioravanzo** [reply]
Thanks a lot for the references. I don't have much time to play around ATM, but I'll report back if I can fix this.

**Reply by StefanoFioravanzo** [reply]
For anyone interested: I just found out that you can use the `~` prefix to show just the last component of the target. It's explained here https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-syntax
